### PR TITLE
mm-api-ref-docs-activity-handler

### PR DESCRIPTION
**Pull request by mmiele**
Added ref docuimentation

### CODE REVIEW

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py [hunk 1]
```diff
@@ -40,9 +57,33 @@ class ActivityHandler:
...
         return
 
     async def on_conversation_update_activity(self, turn_context: TurnContext):
+        """Invoked when a conversation update activity is received from the channel when the base behavior of :meth:`ActivityHandler.on_turn()` is used.
+        
+        :param turn_context: The context object for this turn
+        :type turn_context: :class:`TurnContext`
+        :returns: A task that represents the work queued to execute
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/activity_handler.py]:
```suggestion
        .. note::
```

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py [hunk 2]
```diff
@@ -61,15 +102,63 @@ class ActivityHandler:
...
+        You can add your bot's welcome logic.
+    
+        :param members_added: A list of all the members added to the conversation, as described by the conversation update activity
+        :type members_added: :class:`typing.List`
+        :param turn_context: The context object for this turn
+        :type turn_context: :class:`TurnContext`
+
+        :returns: A task that represents the work queued to execute
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/activity_handler.py]:
```suggestion
        .. note::
```

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py [hunk 3]
```diff
@@ -61,15 +102,63 @@ class ActivityHandler:
...
+        You can add your bot's good-bye logic.
+        
+        :param members_added: A list of all the members removed from the conversation, as described by the conversation update activity
+        :type members_added: :class:`typing.List`
+        :param turn_context: The context object for this turn
+        :type turn_context: :class:`TurnContext`
+
+        :returns: A task that represents the work queued to execute
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/activity_handler.py]:
```suggestion
        .. note::
```

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py [hunk 4]
```diff
@@ -61,15 +102,63 @@ class ActivityHandler:
...
+        Message reactions are only supported by a few channels. The activity that the message reaction corresponds to is indicated in the 
+        reply to Id property. The value of this property is the activity id of a previously sent activity given back to the bot as the response 
+        from a send call.
+        
+        :param turn_context: The context object for this turn
+        :type turn_context: :class:`TurnContext`
+
+        :returns: A task that represents the work queued to execute
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/activity_handler.py]:
```suggestion
        .. note::
```

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py [hunk 5]
```diff
@@ -83,14 +172,68 @@ class ActivityHandler:
...
+        are added to the conversation.
+        
+        :param message_reactions: The list of reactions added
+        :type message_reactions: :class:`typing.List`
+        :param turn_context: The context object for this turn
+        :type turn_context: :class:`TurnContext`
+
+        :returns: A task that represents the work queued to execute
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/activity_handler.py]:
```suggestion
        .. note::
```

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py [hunk 6]
```diff
@@ -83,14 +172,68 @@ class ActivityHandler:
...
+        are removed from the conversation.
+        
+        :param message_reactions: The list of reactions removed
+        :type message_reactions: :class:`typing.List`
+        :param turn_context: The context object for this turn
+        :type turn_context: :class:`TurnContext`
+
+        :returns: A task that represents the work queued to execute
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/activity_handler.py]:
```suggestion
        .. note::
```

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py [hunk 7]
```diff
@@ -83,14 +172,68 @@ class ActivityHandler:
...
 
     async def on_event_activity(self, turn_context: TurnContext):
+        """Invoked when an event activity is received from the connector when the base behavior of :meth:'ActivityHandler.on_turn()` is used.
+       
+        :param turn_context: The context object for this turn
+        :type turn_context: :class:`TurnContext`
+
+        :returns: A task that represents the work queued to execute
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/activity_handler.py]:
```suggestion
        .. note::
```

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py [hunk 8]
```diff
@@ -99,19 +242,63 @@ class ActivityHandler:
...
+        """Invoked when an event other than `tokens/response` is received when the base behavior of
+        :meth:'ActivityHandler.on_event_activity()` is used.
+        This method could optionally be overridden if the bot is meant to handle miscellaneous events.
+        
+        :param turn_context: The context object for this turn
+        :type turn_context: :class:`TurnContext`
+
+        :returns: A task that represents the work queued to execute
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/activity_handler.py]:
```suggestion
        .. note::
```

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py [hunk 9]
```diff
@@ -99,19 +242,63 @@ class ActivityHandler:
...
     ):
+        """Invoked when a `tokens/response` event is received when the base behavior of :meth:'ActivityHandler.on_event_activity()` is used.
+        If using an `oauth_prompt`, override this method to forward this activity to the current dialog.
+        
+        :param turn_context: The context object for this turn
+        :type turn_context: :class:`TurnContext`
+
+        :returns: A task that represents the work queued to execute
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/activity_handler.py]:
```suggestion
        .. note::
```

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py [hunk 10]
```diff
@@ -99,19 +242,63 @@ class ActivityHandler:
...
+        """Invoked when an activity other than a message, conversation update, or event is received when the base behavior of
+        :meth:`ActivityHandler.on_turn()` is used.
+        If overridden, this method could potentially respond to any of the other activity types.
+        
+        :param turn_context: The context object for this turn
+        :type turn_context: :class:`TurnContext`
+
+        :returns: A task that represents the work queued to execute
+
+        .. remarks::
```
Comment (emgrol) [inline libraries/botbuilder-core/botbuilder/core/activity_handler.py]:
```suggestion
        .. note::
```

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py (hunk 1 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/activity_handler.py]: Fixed

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py (hunk 2 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/activity_handler.py]: Fixed

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py (hunk 3 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/activity_handler.py]: Fixed

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py (hunk 4 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/activity_handler.py]: Fixed

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py (hunk 5 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/activity_handler.py]: Fixed

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py (hunk 6 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/activity_handler.py]: Fixed

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py (hunk 7 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/activity_handler.py]: Fixed

Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/activity_handler.py]: Fixed

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py (hunk 8 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/activity_handler.py]: Fixed

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py (hunk 9 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/activity_handler.py]: Fixed

File: libraries/botbuilder-core/botbuilder/core/activity_handler.py (hunk 10 above)
Comment (mmiele) [reply libraries/botbuilder-core/botbuilder/core/activity_handler.py]: Fixed

**Pull request merged**